test/test_ops.py

```diff
@@ -160,6 +160,43 @@ def test_numpy_ref(self, device, dtype, op):
         finally:
             torch.set_default_dtype(cur_default)
 
+    # Tests that the cpu and gpu results are consistent
+    @onlyCUDA
+    @suppress_warnings
+    @ops(_ops_and_refs)
+    def test_compare_cpu(self, device, dtype, op):
+
+        if dtype not in op.dtypes:
+            raise unittest.SkipTest("This test requires both cuda and cpu support")
+        if dtype not in op.dtypesIfCUDA:
+            raise unittest.SkipTest("This test requires both cuda and cpu support")
+
+        def to_cpu(arg):
+            if isinstance(arg, torch.dtype):
+                return arg
+            return arg.to(device='cpu')
+
+        samples = op.sample_inputs(device, dtype)
+
+        for sample in samples:
+            cpu_sample = sample.transform(to_cpu)
+            cuda_results = op(sample.input, *sample.args, **sample.kwargs)
+            cpu_results = op(cpu_sample.input, *cpu_sample.args, **cpu_sample.kwargs)
+
+            self.assertEqual(type(cuda_results), type(cpu_results))
```

Comment:
This may want to call `assert_close` directly:

https://pytorch.org/docs/master/testing.html#torch.testing.assert_close

Which will verify the same type is returned